Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
Overall, I can suggest next changes:
1. When a gun shoots it creates a Projectile object.
2. The projectile `Process` function which called every `GetFreq()` ticks handles the projectile movement.
3. When the projectile cannot move it calls `AttackBy(<projectile_id>)` on the blocker (e.g., human) and `Delete` itself.